--- a/configure	2016-11-18 07:24:06.000000000 +0900
+++ b/configure	2017-04-26 18:04:22.513505316 +0900
@@ -210,6 +210,15 @@
     } elsif ($opt eq '--without-lua_resty_websocket') {
         $resty_opts{no_lua_resty_websocket} = 1;
 
+    } elsif ($opt eq '--without-lua_resty_cookie') {
+        $resty_opts{no_lua_resty_cookie} = 1;
+
+    } elsif ($opt eq '--without-lua_resty_hmac') {
+        $resty_opts{no_lua_resty_hmac} = 1;
+
+    } elsif ($opt eq '--without-lua_resty_http_simple') {
+        $resty_opts{no_lua_resty_http_simple} = 1;
+
     } elsif ($opt eq '--without-lua_resty_lock') {
         $resty_opts{no_lua_resty_lock} = 1;
 
@@ -397,6 +406,10 @@
 
 sub auto_complete ($) {
     my $name = shift;
+    if (-e $name) {
+        return $name;
+    }
+
     my @dirs = glob "$name-[0-9]*" or
         die "No source directory found for $name\n";
 
@@ -949,6 +962,7 @@
         }
 
         for my $key (qw(dns memcached redis mysql string upload websocket
+                        cookie hmac http_simple
                         lock lrucache core upstream_healthcheck limit_traffic))
         {
             unless ($opts->{"no_lua_resty_$key"}) {
@@ -1110,6 +1124,9 @@
                                      disable the lua-resty-upstream-healthcheck library
   --without-lua_resty_string         disable the lua-resty-string library
   --without-lua_resty_websocket      disable the lua-resty-websocket library
+  --without-lua_resty_cookie         disable the lua-resty-cookie library
+  --without-lua_resty_hmac           disable the lua-resty-hmac library
+  --without-lua_resty_http_simple    disable the lua-resty-http-simple library
   --without-lua_resty_limit_traffic  disable the lua-resty-limit-traffic library
   --without-lua_resty_lock           disable the lua-resty-lock library
   --without-lua_resty_lrucache       disable the lua-resty-lrucache library
